ingest/esimmon/cli/watch/__init__.py

```diff
@@ -417,38 +424,46 @@ async def update_range_metadata(gc, image_tarball, bp_path, variable_items, sema
                 attrs = json.loads(attrs[0])
                 vars = fh.available_variables()
 
-                if attrs["type"] != "mesh-colormap":
-                    # All plots except mesh-colormap have x and y vars to grab
+                if y_names := attrs.get("y", None):
+                    y_names = [y_names] if not isinstance(y_names, list) else y_names
+                    new_meta["y_range"] = item_meta.get(
+                        "y_range", [INFINITY, -INFINITY]
+                    )
+                    for y_name in y_names:
+                        new_meta["y_range"] = _update_range(
+                            new_meta["y_range"], vars[y_name]
+                        )
+
+                if x_attr := attrs.get("x", None):
                     new_meta["x_range"] = item_meta.get(
                         "x_range", [INFINITY, -INFINITY]
                     )
-                    x0 = float(vars[attrs["x"]]["Min"])
-                    x1 = float(vars[attrs["x"]]["Max"])
-                    old_x0, old_x1 = new_meta["x_range"]
-                    new_meta["x_range"] = [min(old_x0, x0), max(old_x1, x1)]
+                    new_meta["x_range"] = _update_range(
+                        new_meta["x_range"], vars[x_attr]
+                    )
+
+                if color_attr := attrs.get("color", None):
+                    new_meta["color_range"] = item_meta.get(
+                        "color_range", [INFINITY, -INFINITY]
+                    )
+                    new_meta["color_range"] = _update_range(
+                        new_meta["color_range"], vars[color_attr]
+                    )
 
+                if y_attr := attrs.get("nodes", None):
+                    new_meta["x_range"] = item_meta.get(
```

Comment:
x for y?

Reply:
`nodes` is a 2d mesh that we're pulling both x and y ranges from so maybe `xy_attrs` would be more appropriate. This reminds me though that for one of the plot types we re-use the mesh for all time steps so we shouldn't need to check and set the metadata once it has been set. I'll confirm and add that check.

Reply:
Renamed `y_attr` -> `xy_attrs`. Also added a check for plots of type `mesh-colormap` where the same mesh is used for every time step.